### ISSUE TITLE
Idea: Include Algolia Places example in checkout

### ISSUE THREAD

**Issue by hakonkrogh** [CrystallizeAPI/crystallize-nextjs-boilerplate#76]
The checkout is currently quite barebone in the fields where we record customer information. We could leverage the Algolia Places API here to easily get the users address.

Requirements for setup

- Algolia Javascript Client SDK, lazy loaded in the Checkout
- Algolia App ID (provided by the user)
- Algolia Places API Key (provided by the user)
- Extending the shared `customerInfo` object with the address info given by the search result

**Comment by webmaeistro** [comment]
_Vipps got a deal with VISA recently - could be that soon one can hydrate from Vipps API using just their phone nr (providing they have the app ofc)_

**Comment by domtaylor** [comment]
@webmaeistro not everyone enabling Vipps gateway though

**Comment by webmaeistro** [comment]
@domtaylor that is true... im a Norwegian and as a nation we have really taken vipps to heart old, and young alike..  guess its some bias from my part, ill be the first to admit that..

But the system of having the card holder or your bank send your info apon your approval to do so pr. purchase could be (and should be) implemented - more so these covid-19 days. but i feel we are digressing :D

**Comment by domtaylor** [comment]
@webmaeistro it's the first time I'm hearing about Vipps tbh but glad to hear it's widely adopted in Norway. Klarna has widespread EU adoption but I'm not a huge fan tbh. 

I've never used Vipps as not Norweigan or in Norway, but struggling to see the difference between that and 3D secure offered by Stripe etc.

**Comment by domtaylor** [comment]
Anyway, I've made a start on this, not sure If I'll have much time to finish it off, few issues to sort and very much WIP. If anyone wants to pick up and improve upon - most welcome!

https://github.com/domtaylor/crystallizeAPI-NextJS-boilerplate-AlgoliaPlaces/

**Comment by hakonkrogh** [comment]
> Anyway, I've made a start on this, not sure If I'll have much time to finish it off, few issues to sort and very much WIP. If anyone wants to pick up and improve upon - most welcome!
> 
> https://github.com/domtaylor/crystallizeAPI-NextJS-boilerplate-AlgoliaPlaces/

Great @domtaylor!. There are a lot of changed files due to what I can assume is different `prettier` config. If you can adjust the branch by excluding these files we can add it here as a pull request.

**Comment by Plopix** [comment]
hello here, in order to simplify maintenance of all our boilerplates, we are now using a Mono Repository on which contributions (Issues, and PRs) must be done. It's here: https://github.com/CrystallizeAPI/boilerplates.

This issue should be reopened there, if that still makes sense ;-)

**Comment by AbolfazlAdhami** [comment]
This issue should be reopened there, if that still makes sense ;-)
